action.yml

```diff
@@ -30,9 +34,16 @@ runs:
         else
           URL="https://api.github.com/repos/jandelgado/gcov2lcov/releases/tags/${{inputs.version}}"
         fi
+        if [ -n "${{ inputs.github-token }}" ]; then
+          echo "Using authenticated GitHub API request"
+          PACKAGE_URL=$(curl -H "Authorization: Bearer ${{ inputs.github-token }}" "$URL" | jq -r '[ .assets[] | select(.name | test("gcov2lcov.*linux.amd64.tar.gz")) | .browser_download_url ] | first ')
```

Comment:
DRY: set a `HEADER` variable here to `-H "Authorization: ..."`  if the condition is met and use it below. By default, `HEADER` is empty. Don't have duplicate curl and jq invocations.

Reply:
Thank you for the review, updated.